src/components/Editor/components/Main.js

```diff
@@ -28,12 +31,34 @@ class Main extends React.Component {
     }
   }
 
+  resetSaveText = () => {
+    this.setState({
+      saveText: "Save code",
+    });
+  };
+
+  handleSave = event => {
+    var programsJson = {};
+
+    programsJson["HTML"] = this.props.programs.getIn(["HTML", "code"]);
+    programsJson["Processing"] = this.props.programs.getIn(["Processing", "code"]);
+    programsJson["Python"] = this.props.programs.getIn(["Python", "code"]);
+
+    fetch.updatePrograms(this.props.uid, programsJson).then(() => {
```

Comment:
For now I think this is fine, but we should change this to check if there was an error/it failed to save. Maybe we can make the button red and make the text 'Failed to save'